File: sdks/typescript/src/resources/admin/audit.ts
import type { RequestClient } from "../../core.js";
import type { AuditLogListResponse } from "../../types/index.js";
import { pagination } from "./_shared.js";

export class AdminAuditResource {
  constructor(private readonly _client: RequestClient) {}

  list(
    options: {
      action?: string;
      actorId?: string;
      resourceType?: string;
      limit?: number;
      offset?: number;
    } = {},
  ): Promise<AuditLogListResponse> {
    const params = pagination(options);
    if (options.action !== undefined) params.action = options.action;
    if (options.actorId !== undefined) params.actor_id = options.actorId;
    if (options.resourceType !== undefined) params.resource_type = options.resourceType;
    return this._client._request("GET", "/v1/admin/audit", { params });
  }
}

File: sdks/python/src/bigrag/resources/admin/audit.py
from __future__ import annotations

from typing import TYPE_CHECKING

from bigrag.resources.admin._shared import _pagination
from bigrag.types.admin import AuditLogListResponse

if TYPE_CHECKING:
    from bigrag._core import BigRAGCore


class AdminAuditResource:
    def __init__(self, client: BigRAGCore) -> None:
        self._client = client

    async def list(
        self,
        *,
        action: str | None = None,
        actor_id: str | None = None,
        resource_type: str | None = None,
        limit: int | None = None,
        offset: int | None = None,
    ) -> AuditLogListResponse:
        params = _pagination(limit=limit, offset=offset)
        if action is not None:
            params["action"] = action
        if actor_id is not None:
            params["actor_id"] = actor_id
        if resource_type is not None:
            params["resource_type"] = resource_type
        return await self._client._request("GET", "/v1/admin/audit", params=params)
